src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -200,9 +200,22 @@ private static int halfMaxSize(Collection<CaseResult> results) {
      * @param testName Test name.
      * @param errorStackTrace Error stack trace.
      */
+    @Deprecated
     public CaseResult(SuiteResult parent, String testName, String errorStackTrace) {
+        this(parent, testName, errorStackTrace, null);
+    }
+    
+    /**
+     * Used to create a fake failure, when Hudson fails to load data from XML files.
+     *
+     * @param parent Parent result.
+     * @param testName Test name.
+     * @param errorStackTrace Error stack trace.
+     * @since FIXME
+     */
+    public CaseResult(SuiteResult parent, String testName, String errorStackTrace, String testRunName) {
```

Comment:
🐜  Make use of CheckForNull and NonNull here.   Javadocs leave something to be desired.  Like content.